store/store.go

```diff
@@ -0,0 +1,114 @@
+package store
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+	"user-favorites-api/models"
+
+	"github.com/dgrijalva/jwt-go"
+	"github.com/jackc/pgx/v5/pgxpool"
+	"golang.org/x/crypto/bcrypt"
+)
+
+var JwtKey = []byte(os.Getenv("SECRET_KEY"))
+
+type Claims struct {
```

Comment:
What approach would you choose to avoid including JWT definitions and processes in the repository layer?

Reply:
I would have used a service layer approach so I could adhere to the single responsibility principle. This pattern, utilized in many frameworks, including NestJs which I am familiar with, enforces good design practices.

- All JWT-related operations would have been moved to a separate file.
- A new file containing a service would have been responsible for handling the business logic.
- I would have retained the `store/store.go`, which in my case represents the repository layer, for database-related operations only.